presto-main/src/main/java/io/prestosql/server/security/CertificateAuthenticatorManager.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.security;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.prestosql.spi.security.CertificateAuthenticator;
+import io.prestosql.spi.security.CertificateAuthenticatorFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.Maps.fromProperties;
+import static io.prestosql.util.PropertiesUtil.loadProperties;
+import static java.util.Objects.requireNonNull;
+
+public class CertificateAuthenticatorManager
+{
+    private static final Logger log = Logger.get(CertificateAuthenticatorManager.class);
+
+    private static final File CONFIG_FILE = new File("etc/certificate-authenticator.properties");
+    private static final String NAME_PROPERTY = "certificate-authenticator.name";
+
+    private final AtomicBoolean required = new AtomicBoolean();
+    private final Map<String, CertificateAuthenticatorFactory> factories = new ConcurrentHashMap<>();
+    private final AtomicReference<io.prestosql.spi.security.CertificateAuthenticator> authenticator = new AtomicReference<>();
+
+    public void setRequired()
+    {
+        required.set(true);
+    }
+
+    public void addCertificateAuthenticatorFactory(CertificateAuthenticatorFactory factory)
+    {
+        checkArgument(factories.putIfAbsent(factory.getName(), factory) == null,
+                "Certificate authenticator '%s' is already registered", factory.getName());
+    }
+
+    public void loadCertificateAuthenticator()
+            throws Exception
+    {
+        Map<String, String> properties;
+        File configFile;
+        String name;
+
+        if (!required.get()) {
+            return;
+        }
+
+        configFile = CONFIG_FILE.getAbsoluteFile();
+        try {
+            properties = new HashMap<>(loadProperties(configFile));
+        }
+        catch (FileNotFoundException e) {
```

Comment:
I don't think we should have a fallback. We can require explicit configuration of the authenticator. Relying on file existence is error prone, as a user might have wanted to configure one, but named the file incorrectly. It seems better to be consistent.

@dain thoughts?